spring-boot-starter/starter-security/src/main/java/org/camunda/bpm/spring/boot/starter/security/oauth2/impl/OAuth2IdentityProvider.java

```diff
@@ -0,0 +1,395 @@
+    return springSecurityAuthentication() ? null : super.findTenantById(tenantId);
+  }
+
+  @Override
+  public TenantQuery createTenantQuery() {
+    return springSecurityAuthentication() ? new OAuth2TenantQuery() : super.createTenantQuery();
+  }
+
+  @Override
+  public TenantQuery createTenantQuery(CommandContext commandContext) {
+    return springSecurityAuthentication() ? new OAuth2TenantQuery() : super.createTenantQuery();
+  }
+
+  @Override
+  public void flush() {
+    if (!springSecurityAuthentication()) {
+      super.flush();
+    } // else nothing to do
+  }
+
+  @Override
+  public void close() {
+    if (!springSecurityAuthentication()) {
+      super.flush();
+    } // else nothing to do
+  }
+
+  // WriteableIdentityProvider methods
+
+  @Override
+  public UserEntity createNewUser(String userId) {
+    if (springSecurityAuthentication()) {
+      unsupportedOperationForOAuth2();
+    }
+    return super.createNewUser(userId);
+  }
+
+  @Override
+  public IdentityOperationResult saveUser(User user) {
+    if (springSecurityAuthentication()) {
+      unsupportedOperationForOAuth2();
+    }
+    return super.saveUser(user);
+  }
+
+  @Override
+  public IdentityOperationResult deleteUser(String userId) {
+    if (springSecurityAuthentication()) {
+      unsupportedOperationForOAuth2();
+    }
+    return super.deleteUser(userId);
+  }
+
+  @Override
+  public IdentityOperationResult unlockUser(String userId) {
+    if (springSecurityAuthentication()) {
+      unsupportedOperationForOAuth2();
+    }
+    return super.unlockUser(userId);
+  }
+
+  @Override
+  public GroupEntity createNewGroup(String groupId) {
+    if (springSecurityAuthentication()) {
+      unsupportedOperationForOAuth2();
+    }
+    return super.createNewGroup(groupId);
+  }
+
+  @Override
+  public IdentityOperationResult saveGroup(Group group) {
+    if (springSecurityAuthentication()) {
+      unsupportedOperationForOAuth2();
+    }
+    return super.saveGroup(group);
+  }
+
+  @Override
+  public IdentityOperationResult deleteGroup(String groupId) {
+    if (springSecurityAuthentication()) {
+      unsupportedOperationForOAuth2();
+    }
+    return super.deleteGroup(groupId);
+  }
+
+  @Override
+  public Tenant createNewTenant(String tenantId) {
+    if (springSecurityAuthentication()) {
+      unsupportedOperationForOAuth2();
+    }
+    return super.createNewTenant(tenantId);
+  }
+
+  @Override
+  public IdentityOperationResult saveTenant(Tenant tenant) {
+    if (springSecurityAuthentication()) {
+      unsupportedOperationForOAuth2();
+    }
+    return super.saveTenant(tenant);
+  }
+
+  @Override
+  public IdentityOperationResult deleteTenant(String tenantId) {
+    if (springSecurityAuthentication()) {
+      unsupportedOperationForOAuth2();
+    }
+    return super.deleteTenant(tenantId);
+  }
+
+  @Override
+  public IdentityOperationResult createMembership(String userId, String groupId) {
```

Comment:
❌ Enable membership operations for oauth2 as well so authorized users can create new authorizations for other users.

Reply:
Why do we need membership operations? We don't allow them in LDAP either. Information about a user's group membership comes from the identity provider.

Reply:
The initial comment was added after we discussed whether a user should be able to add authorizations in Camunda. I double checked, and with and without the provider, users can add authorizations. So I don't think any action is needed here.
The code here is about group membership and I think Tassilo's right. It should behave the same as LDAP.